app/models/concerns/act_as_taggable_on.rb

```diff
@@ -1,25 +1,21 @@
 module ActAsTaggableOn
   def acts_as_taggable_on
     class_eval do
+      require "hair_trigger"
+
       def self.tagging_relation_name
-        "#{name.underscore}_tags".to_sym
+        "taggings".to_sym
       end
 
-      has_many tagging_relation_name
+      has_many tagging_relation_name, :as => :resource, :inverse_of => :resource
       has_many :tags, :through => tagging_relation_name
 
       def self.taggable?
         true
       end
 
-      def taggings
-        public_send(self.class.tagging_relation_name).map do |tagging|
-          {
-            :tag_id => tagging.tag_id.to_s,
-            :name   => tagging.tag.name,
-            :value  => tagging.value,
-          }
-        end
+      trigger.after(:delete) do
+        "DELETE FROM taggings WHERE resource_type='#{name}' AND resource_id=OLD.id"
```

Comment:
What does this do? We should already have the DB trigger, right?

Reply:
Not if we go to a single taggings table or collapse taggings and tags into just tags.